pulsar-client-all/pom.xml

```
@@ -397,6 +397,33 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <version>${maven-enforcer-plugin.version}</version>
+        <executions>
+          <execution>
+            <id>enforce-bytecode-version</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <enforceBytecodeVersion>
+                  <maxJdkVersion>1.8</maxJdkVersion>
```

I guess it would be consistent to use the `pulsar.client.compiler.release` property
```suggestion
                  <maxJdkVersion>${pulsar.client.compiler.release}</maxJdkVersion>
```